File: src/Calculator/Converters/BooleanNegationConverter.cpp
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

#include "pch.h"
#include "BooleanNegationConverter.h"

using namespace CalculatorApp::Converters;

using namespace Platform;
using namespace Windows::Foundation;
using namespace Windows::UI::Xaml::Interop;

Object ^ BooleanNegationConverter::Convert(Object ^ value, TypeName targetType, Object ^ parameter, String ^ language)
{
    (void)targetType; // Unused parameter
    (void)parameter;  // Unused parameter
    (void)language;   // Unused parameter

    auto boxedBool = dynamic_cast<Box<bool> ^>(value);
    auto boolValue = (boxedBool != nullptr && boxedBool->Value);
    return !boolValue;
}

Object ^ BooleanNegationConverter::ConvertBack(Object ^ value, TypeName targetType, Object ^ parameter, String ^ language)
{
    (void)targetType; // Unused parameter
    (void)parameter;  // Unused parameter
    (void)language;   // Unused parameter

    auto boxedBool = dynamic_cast<Box<bool> ^>(value);
    auto boolValue = (boxedBool != nullptr && boxedBool->Value);
    return !boolValue;
}

File: src/Calculator/Converters/BooleanNegationConverter.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

using System;

namespace CalculatorApp
{
    namespace Converters
    {
        /// <summary>
        /// Value converter that translates true to false and vice versa.
        /// </summary>
        [Windows.Foundation.Metadata.WebHostHidden]
        public sealed class BooleanNegationConverter : Windows.UI.Xaml.Data.IValueConverter
        {
            public object Convert(object value, Type targetType, object parameter, string language)
            {
                var boolValue = (value is bool boxedBool && boxedBool);
                return !boolValue;
            }

            public object ConvertBack(object value, Type targetType, object parameter, string language)
            {
                var boolValue = (value is bool boxedBool && boxedBool);
                return !boolValue;
            }
        }
    }
}

